### PR TITLE
[SDK-549] Add public user id getter

### PR DESCRIPTION
**Pull request by alessioborraccino**
## Description

Adds a public way to get userID from the client.

## Motivation and Context

Raising the need to tag resources with something related to the user account, we decided to make the userId available from fetching from outside.

## How is it being implemented?

Adds a new API from the client to get the user ID which fails in case user is not logged in.

## How Has This Been Tested?

Adds new tests for the new api calls:
* UserService unit tests
* Data4LifeClient unit tests
* Data4LifeClient + UserService module tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: SDK/Sources/Client/Data4LifeClient+User.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    public func sessionStateDidChange(queue: DispatchQueue = responseQueue,
+                                      completion: @escaping (Bool) -> Void) {
+        guard oAuthService.sessionStateChanged == nil else { return }
+        oAuthService.sessionStateChanged = completion
+    }
+
+    /**
+     Returns the userID connected with the account, only if the account is logged in, returns a Not Logged In error otherwise.
+
+     - parameter completion:Completion that returns the userID connected with the account.
```
Comment (jose-cuadros-data4life) [inline SDK/Sources/Client/Data4LifeClient+User.swift:115]: Just a small fix: `completion: Completion`

File: SDK/Sources/Services/UserService.swift [hunk 2]
```diff
@@ -64,4 +65,8 @@ class UserService: UserServiceType {
             try wait(self.keychainService.set(response.userId, forKey: .userId))
         }
     }
+
+    func getUserId() throws -> String {
+        try `await`(self.keychainService.get(.userId))
```
Comment (jose-cuadros-data4life) [inline SDK/Sources/Services/UserService.swift:70]: We could use `wait` from the other PR

File: SDK/Tests/ModuleTests/Data4LifeClient+UserServiceModuleTests.swift [hunk 3]
```diff
@@ -0,0 +1,319 @@
+//  Copyright (c) 2020 D4L data4life gGmbH
+//  All rights reserved.
```
Comment (jose-cuadros-data4life) [inline SDK/Tests/ModuleTests/Data4LifeClient+UserServiceModuleTests.swift:2]: Nice tests in this file! 🎉

**Review by jose-cuadros-data4life — CHANGES_REQUESTED** [review]

**Review by jose-cuadros-data4life — APPROVED** [review]
Lgtm

File: SDK/Tests/ClientTests/Data4LifeClient+UserTests.swift [hunk 4]
```diff
@@ -403,4 +403,31 @@ extension Data4LifeClientUserTests {
         oAuthService.sessionStateChanged?(true)
         waitForExpectations(timeout: 5)
     }
+
+    func testGetUserIdWhenLoggedIn() {
+        let userId = UUID().uuidString
+        userService.getUserIdResult = userId
+
+        let asyncExpectation = expectation(description: "should return response with count on UI thread")
```
Comment (ztepsa) [inline SDK/Tests/ClientTests/Data4LifeClient+UserTests.swift:411]: Can we update expectation description so it reflects what's really going on, something like:
```
should return userId on on UI thread
```
Also we can assert that we'r on the proper thread if needed.
```swift
XCTAssertTrue(Thread.isMainThread)
```

**Pull request merged**